virtcontainers/store/backend.go

```diff
@@ -49,4 +49,6 @@ type backend interface {
 	load(item Item, data interface{}) error
 	store(item Item, data interface{}) error
 	raw(id string) (string, error)
+	lock(item Item, exclusive bool) (string, error)
```

Comment:
I don't get why we need those new locking API. If I followed until then, we have any `Load()` performing a read only lock, and `Store()` performing a read/write lock, which looks like it should be enough. Also, from the Kata API perspective, we have the locking of the pod, meaning we cannot access two containers of the same pod concurrently.

Ohh maybe I understand now, are you trying to do the pod locking through this API? And you're doing this because the lock file has to be created somewhere... It's a tricky one because it's not natural to have the locking being handled through the storage, but I guess that's how it was kind of working already.

Unless we share the global path for pods (in `types` maybe), and we could create a separate `lock/` package.

Reply:
@sameo did you miss this comment?